analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,186 @@
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+#Function to split pathway names from TukeyHSD into vector of 2 elements
+#i.e. Tukey HSD outputs Pathway1-Pathway2 and this will spitout c(Pathway1, Pathway2)
+#Will be used so that we can tell quickly which pathways are consistently upregulated vs other pathways
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2]
+	return(c(tmp1, tmp2))
+}
+
+#Analyse Pathways statistically correlated with clinical variables
+#Returns ANOVA P-value and Tukey HSD Results
+#Requires Pathway (pathwayName) and Clinical Variable (clinVarName)
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar") 	#Need to set this since pathwayName and clinVarName will change which is not good for aov function
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	aovOutNamed <- unlist(summary(aovOut))
+	posthoc[,"F_Value"] <- aovOutNamed["F value1"]
+	posthoc[,"AOV_P"] <- aovOutNamed["Pr(>F)1"]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc) #Pulls out disease names 
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc)
+}
+
+
+#Analysis by short_histology. This is only clinical variable we are examining but the function compareClinVarToPathway
+#is generic enough that any variable in clinical data can be used. This will be useful if we decide to use for instance molecular
+#subtypes instead of histology for this analysis. So the lapply below will for each pathway
+#run an ANOVA across all histologies i.e. looking to see if any particular histology has a significantly greater score
+#than any other histology and then follow that with a post-hoc test. The comparison is not done between pathways. 
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology")
+diseaseType <- do.call("rbind", diseaseType) 
+
+#Filter to significant entries by F and Tukey and write out 
+diseaseTypeFilt <- diseaseType %>% dplyr::filter( AOV_P <= aovParam, p.adj <= padjParam)
+colnames(diseaseTypeFilt)[1:2] <- c("DiseaseX", "DiseaseY") #Rename columns to match clinical variable type
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#We would like to filter to cases where a histology is significantly higher than other histologies (and the corresponding pathways)
+#This is for the heatmap so we may focus a bit on the more interesting results
+#The issue here is that Tukey will write out the pathways in some particular order
```

Comment:
Write out pathways or the clinical variables under consideration?

Reply:
Yeah, here it is histology, updated. Thanks.